pkg/kvstore/etcd.go

```diff
@@ -101,12 +134,40 @@ func NewEtcdClient(config *client.Config, cfgPath string) (KVClient, error) {
 				ec.session = newSession
 				ec.sessionMU.Unlock()
 				log.Debugf("Renewing etcd session")
+				ec.CheckMinVersion()
 			}
 		}
 	}()
 	return ec, nil
 }
 
+// CheckMinVersion checks the minimal version running on etcd cluster. If the
+// minimal version running doesn't meet cilium minimal requirements, exits
+// cilium.
+func (e *EtcdClient) CheckMinVersion() {
+	eps := e.cli.Endpoints()
+	success := false
+	for _, ep := range eps {
+		v, err := getClusterVersionFrom(ep)
+		if err != nil {
+			log.Debugf("Unable to check min version: %s", err)
+			continue
+		}
+		if minEVersion.Check(v) {
+			log.Infof("etcd cluster version: %s", v.String())
+			success = true
+			break
+		}
+		// FIXME: after we rework the refetching IDs for a connection lost
```

Comment:
We could print something like `log.Info("Established connectivity to etcd cluster version  %s")` or return and error and then do the failure handling in the caller and do a Fatal on bootstrap and a warning and retry for the session close case?